src/plugins/vis_type_table/public/table_vis_response_handler.ts

```diff
@@ -30,15 +30,20 @@
 
 import { getFormatService } from './services';
 import { OpenSearchDashboardsDatatable } from '../../expressions/public';
-import { TableVisConfig } from './types';
-
+import { FormattedColumn, TableVisConfig } from './types';
+import { convertToFormattedData } from './utils/convert_to_formatted_data';
 export interface Table {
   columns: OpenSearchDashboardsDatatable['columns'];
   rows: OpenSearchDashboardsDatatable['rows'];
 }
 
+export interface FormattedTable {
+  columns: FormattedColumn[];
+  rows: OpenSearchDashboardsDatatable['rows'];
+}
+
 export interface TableGroup {
-  table: OpenSearchDashboardsDatatable;
+  table?: FormattedTable;
```

Comment:
do you really want this to be optional? Is it valid to have a `TableGroup` with an undefined `table` prop (see Miki's question https://github.com/opensearch-project/OpenSearch-Dashboards/pull/3397/files#r1122178080)?